src/utilities/network-information-utils.ts

```diff
@@ -0,0 +1,84 @@
+import { NavigatorConnection } from "../interfaces/navigator-connection";
+import {
+    MozillaNavigator,
+    StandardNavigator,
+    WebkitNavigator,
+} from "../interfaces/navigator";
+import { NetworkConnection } from "../interfaces/network-connection";
+import { NavigatorConnectionVariants } from "../enumerations/navigator-connection-variants";
+
+// -----------------------------------------------------------------------------------------
+// #region Types
+// -----------------------------------------------------------------------------------------
+
+type WindowNavigator = Navigator &
+    Partial<MozillaNavigator & StandardNavigator & WebkitNavigator>;
+
+// #endregion Types
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+/**
+ * Returns a NavigatorConnection object if one exists
+ */
+const getNavigatorConnection = (): NavigatorConnection | undefined => {
+    if (isStandardNavigator(window.navigator)) {
+        return window.navigator.connection;
+    }
+
+    if (isMozillaNavigator(window.navigator)) {
+        return window.navigator.mozConnection;
+    }
+
+    if (isWebkitNavigator(window.navigator)) {
+        return window.navigator.webkitConnection;
+    }
+
+    return undefined;
+};
+
+/**
+ * Returns a `NetworkConnection` object which is an aggregate of `navigator.connection` and `navigator.onLine`
+ */
+const getNetworkConnection = (): NetworkConnection => {
+    const { onLine: isOnline } = window.navigator;
+    const navigatorConnection = getNavigatorConnection() ?? {};
+
+    return {
+        ...navigatorConnection,
+        isOnline,
+    };
+};
+
+const isMozillaNavigator = (
+    navigator: WindowNavigator
+): navigator is MozillaNavigator => {
+    return navigator[NavigatorConnectionVariants.Mozilla] != null;
+};
+
+const isStandardNavigator = (
+    navigator: WindowNavigator
+): navigator is StandardNavigator => {
+    return navigator[NavigatorConnectionVariants.Standard] != null;
+};
+
+const isWebkitNavigator = (
+    navigator: WindowNavigator
+): navigator is WebkitNavigator => {
+    return navigator[NavigatorConnectionVariants.Webkit] != null;
+};
+
+// #endregion Functions
+
+// -----------------------------------------------------------------------------------------
+// #region Exports
+// -----------------------------------------------------------------------------------------
+
+export const NetworkInformationUtils = {
```

Comment:
Thought - since this a new util, can we restructure the way we're defining it? I believe with this setup (which a lot of the current utils follow), the JSDocs get stripped off from the type definitions. See this issue/comment for further explanation: https://github.com/AndcultureCode/AndcultureCode.JavaScript.Core/issues/51#issuecomment-748977095

Defining `NetworkInformationUtils` up further and those functions on the object seems to retain the docs.

```ts
const NetworkInformationUtils = {
    /**
    * JSDocs 
    */
    getNavigatorConnection() : NetworkConnection {
        // ...
    }
    
// ...

export { NetworkInformationUtils };
```